mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,209 @@
+        sklearn-style model that contains a fit and predict method.
+    """
+    def __init__(self, stim_features, lags, sfreq=1., model=None):
+        self.stim_features = stim_features
+        self.lags = lags
+        self.model = Ridge() if model is None else model
+        self.sfreq = sfreq
+
+    def fit(self, X, y):
+        """Fit a receptive field model.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs], n_channels, n_times)
+            The input features for the model.
+        y : array, shape ([n_epochs], n_times)
+            The output feature for the model.
+        """
+        X_del, msk = self._delay_for_fit(X)
+
+        # Reshape so we have times x features
+        X_del = X_del[..., ~msk]
+        y = y[..., ~msk]
+
+        if X_del.ndim == 4:
+            # We have epochs, so hstack now
+            n_ep = X_del.shape[1]
+            X_del = X_del.swapaxes(0, 1).reshape(n_ep, -1, X_del.shape[-1])
+            X_del = np.hstack(X_del)
+            y = np.hstack(y)
+
+        # Reshape so time is last dimension
+        X_del = X_del.reshape([-1, X_del.shape[-1]])
+        y = y.reshape([-1, X_del.shape[-1]])
+
+        self.model.fit(X_del.T, y.T)
+        self.msk_remove = msk
+
+    def transform(self, X=None, y=None):
+        """Does nothing, used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : None
+            Used for sklearn compatibility.
+        y : None
+            Used for sklearn compatibility.
+        """
+        return self
+
+    def fit_transform(self, X=None, y=None):
+        """Does nothing, used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : None
+            Used for sklearn compatibility.
+        y : None
+            Used for sklearn compatibility.
+        """
+        return self
+
+    def predict(self, X, y=None):
+        """Make predictions using a receptive field.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs], n_channels, n_times)
+            The input features for the model.
+        y : None
+            Used for sklearn compatibility.
+
+        Returns
+        -------
+        y_pred : array, shape (n_times * n_epochs)
+            The output predictions with time concatenated.
+        """
+        X_del, msk = self._delay_for_fit(X)
+        X_del = X_del[..., ~msk]
+        X_del = X_del.reshape([-1, X_del.shape[-1]]).T
+        y_pred = self.model.predict(X_del)
+        return y_pred
+
+    def _delay_for_fit(self, X):
+        # First delay
+        X_del = delay_time_series(
+            X, self.lags, self.sfreq)
+
+        # Mask for removing edges later
+        msk_helper = np.ones(X.shape[-1])
+        msk_helper = delay_time_series(msk_helper, self.lags, self.sfreq)
+        msk = (msk_helper == 0).any(axis=0)
+        return X_del, msk
+
+    def plot_coefs(self, ax=None):
```

Comment:
Well it is a bit convenient because it can automatically set the axes and labels, etc. But we might want to special-case the 1D / time-waveform vs 2D / time-freq-style cases if we generalize to n-dimensions like I mentioned above.